benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp

```diff
@@ -598,124 +599,16 @@ void GraphicsBenchmarkApp::Render()
-                    data.ITModelMatrix              = glm::inverse(glm::transpose(data.modelMatrix));
-                    data.ambient                    = float4(0.3f);
-                    data.cameraViewProjectionMatrix = mCamera.GetViewProjectionMatrix();
-                    data.lightPosition              = float4(mLightPosition, 0.0f);
-                    data.eyePosition                = float4(mCamera.GetEyePosition(), 0.0f);
-                    mDrawCallUniformBuffers[i]->CopyFromSource(sizeof(data), &data);
-
-                    frame.cmd->PushGraphicsUniformBuffer(mSphere.pipelineInterface, /* binding = */ 0, /* set = */ 0, /* bufferOffset = */ 0, mDrawCallUniformBuffers[i]);
-                    frame.cmd->PushGraphicsSampledImage(mSphere.pipelineInterface, /* binding = */ 1, /* set = */ 0, mAlbedoTexture.sampledImageView);
-                    frame.cmd->PushGraphicsSampler(mSphere.pipelineInterface, /* binding = */ 2, /* set = */ 0, mAlbedoTexture.sampler);
-                    frame.cmd->PushGraphicsSampledImage(mSphere.pipelineInterface, /* binding = */ 3, /* set = */ 0, mNormalMapTexture.sampledImageView);
-                    frame.cmd->PushGraphicsSampler(mSphere.pipelineInterface, /* binding = */ 4, /* set = */ 0, mNormalMapTexture.sampler);
-                    frame.cmd->PushGraphicsSampledImage(mSphere.pipelineInterface, /* binding = */ 5, /* set = */ 0, mMetalRoughnessTexture.sampledImageView);
-                    frame.cmd->PushGraphicsSampler(mSphere.pipelineInterface, /* binding = */ 6, /* set = */ 0, mMetalRoughnessTexture.sampler);
-
-                    uint32_t indexCount = indicesPerDrawCall;
-                    // Add the remaining indices to the last drawcall
-                    if (i == currentDrawCallCount - 1) {
-                        indexCount += (currentSphereCount * mSphereIndexCount - currentDrawCallCount * indicesPerDrawCall);
-                    }
-                    uint32_t firstIndex = i * indicesPerDrawCall;
-                    frame.cmd->DrawIndexed(indexCount, /* instanceCount = */ 1, firstIndex);
-                }
-            }
-        }
-        frame.cmd->EndRenderPass();
-
-        // =====================================================================
-        // Fullscreen quads renderpasses
-        // =====================================================================
-        if (pFullscreenQuadsCount->GetValue() > 0) {
-            frame.cmd->BindGraphicsPipeline(mFullscreenQuads.pipeline);
-            frame.cmd->BindVertexBuffers(1, &mFullscreenQuads.vertexBuffer, &mFullscreenQuads.vertexBinding.GetStride());
-
-            for (size_t i = 0; i < pFullscreenQuadsCount->GetValue(); ++i) {
-                currentRenderPass = swapchain->GetRenderPass(imageIndex);
-                PPX_ASSERT_MSG(!currentRenderPass.IsNull(), "render pass object is null");
+        // Note: Transitions image layout PRESENT->RENDER before this first renderpass
```

Comment:
Do you think it would make sense to take those Transition lines out of Render* functions and just put here and remove this comment? That way you don't have to document it here with the comment, and if you add more Render* functions you don't have to move the image layout code. RenderGUI would also not need to have image layout transition code regardless if imgui is on or not

Reply:
Acknowledged that this is unintuitive, I have changed this.